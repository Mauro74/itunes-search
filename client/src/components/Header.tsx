import styled from "styled-components";

const SiteHeader = styled.header`
  padding: 0.5rem 0;
  text-align: center;
  border-bottom: solid 1px var(--light-grey);
  background-color: var(--white);
`;

const Header = (): JSX.Element => {
  return (
    <SiteHeader>
      <h1>iTunes Search</h1>
    </SiteHeader>
  );
};

export default Header;
