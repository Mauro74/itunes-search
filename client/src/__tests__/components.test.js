import * as React from "react";
import * as renderer from "react-test-renderer";
import ProductComponent from "../components/ProductList";
import ProductListing from "../components/ProductSearch";
import Header from "../components/Header";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

describe("components", () => {
  const initialState = {};
  const mockStore = configureStore();
  let store;

  test("ProductComponent should render correctly", () => {
    store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <ProductComponent />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("ProductListing should render correctly", () => {
    store = mockStore(initialState);
    const tree = renderer
      .create(
        <Provider store={store}>
          <ProductListing />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Header should render correctly", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
