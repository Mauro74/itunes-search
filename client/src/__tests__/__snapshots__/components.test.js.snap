// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Header should render correctly 1`] = `
<header
  className="sc-eDvSVe iUNhIs"
>
  <h1>
    iTunes Search
  </h1>
</header>
`;

exports[`components ProductComponent should render correctly 1`] = `
<h5
  className="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter"
>
  No results for this search
</h5>
`;

exports[`components ProductListing should render correctly 1`] = `
<main
  className="sc-gswNZR sepQv"
>
  <div
    className="sc-dkrFOg giFhjA"
  >
    <h2>
      Search
    </h2>
    <form
      className="sc-hLBbgP bfUVLx"
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiTextField-root"
      >
        <div
          className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiFilledInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="enter search term"
            required={false}
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          Search
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Clear
        </span>
      </button>
    </form>
  </div>
</main>
`;
